### PR TITLE
Spanner: use WatchdogProvider

### PR DESCRIPTION
**Pull request by yihanzhen**
Switch from `WatchdogInterceptor` to `WatchdogProvider`, which is one of the shared tools implemented in gax-java. This is non-breaking change and comes with the gapic migration work, as we aim to have spanner maximize the use of shared tools after the migration.

`WatchdogInterceptor` was to monitor streaming calls every `periodSeconds` and once the client side has waited for too long or the stream has become idle for too long, the stream would be terminated. The timeouts are derived from system properties, and if not specified default values will be provided.

Change to use [`WatchdogProvider`](https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/rpc/WatchdogProvider.java) preserves the same behaviors (same `periodSeconds`, which is renamed to `checkInterval` and same `waitTimeout` and `idleTimeout`, which are put in `GrpcCallContext`).

### CODE REVIEW

**Comment by hzyi-google** [comment]
@nithinsujir @garrettjonesgoogle @snehashah16 PTAL

**Comment by hzyi-google** [comment]
@nithinsujir updated. PTAL.

File: google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java [hunk 1]
```diff
@@ -173,6 +191,17 @@ public GapicSpannerRpc(SpannerOptions options) {
     CredentialsProvider credentialsProvider =
         GrpcTransportOptions.setUpCredentialsProvider(options);
 
+    WatchdogProvider watchdogProvider =
+        InstantiatingWatchdogProvider.create()
+            .withExecutor(
+                Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder()
+                        .setDaemon(true)
+                        .setNameFormat("Cloud-Spanner-WatchdogInterceptor-%d")
```
Comment (nithinsujir) [inline google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java]: Should this be WatchdogProvider?

Comment (hzyi-google) [reply google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java]: Yes! Nice catch! Thanks.

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/WatchdogInterceptorTest.java [hunk 2]
```diff
@@ -1,202 +0,0 @@
-/*
```
Comment (snehashah16) [inline google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/WatchdogInterceptorTest.java]: Can this test be refactored to test new functionality ?

Comment (hzyi-google) [reply google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/WatchdogInterceptorTest.java]: I thought about that, but these tests are all unit tests with mocks and the [test](https://github.com/googleapis/gax-java/blob/master/gax/src/test/java/com/google/api/gax/rpc/WatchdogTest.java) for watchdog in gax is mostly equivalent so it seems unnecessary.

**Pull request merged**